src/routes/videoPlaybackProxy.ts

```diff
@@ -168,9 +210,9 @@ videoPlaybackProxy.get("/", async (c) => {
         }
     }
 
-    return new Response(googlevideoResponse.body, {
+    return new Response(stream.responseReadable, {
         status: responseStatus,
-        statusText: googlevideoResponse.statusText,
+        statusText: headResponse.statusText,
         headers: headersForResponse,
     });
```

Comment:
finally, the headers from the `HEAD` response are sent back and the `POST` is only used for the streaming data response. Seems to work just fine. Will need some decently thorough testing.